### PR TITLE
Fix ANOVA issue with multiple covariates and no fixed factors

### PR DESCRIPTION
**Pull request by JohnnyDoorn**

### CODE REVIEW

File: JASP-Engine/JASP/R/ancova.R [hunk 1]
```diff
@@ -856,7 +856,10 @@ Ancova <- function(dataset=NULL, options, perform="run", callback=function(...)
 				F <- if (is.na(result[term,"F value"])) {""} else { result[term, "F value"] }
 				p <- if (is.na(result[term,"Pr(>F)"] )) {""} else { result[term, "Pr(>F)"] }
 
-				if(i == 1 || term == "Residuals" || (!is.null(unlist(options$covariates)) && (terms.normal[i] == options$fixedFactors[[1]] || terms.normal[i] == options$covariates[[1]]) && 
+				firstFixedFactor <- ifelse(length(options$fixedFactors) == 0, "", options$fixedFactors[[1]])
```
Comment (AlexanderLyNL) [inline JASP-Engine/JASP/R/ancova.R]: Can you write this in an if-else statement. The ifelse construction can lead to some weird behaviour. That won't occur here though, but it's more a matter of style

Comment (AlexanderLyNL) [reply JASP-Engine/JASP/R/ancova.R]: In addition, could you add a unit test?

**Comment by AlexanderLyNL** [comment]
Other than that the fix works like a charm.

File: JASP-Engine/JASP/R/ancova.R (hunk 1 above)
Comment (JohnnyDoorn) [reply JASP-Engine/JASP/R/ancova.R]: What unit test? There is already one for ANCOVA right?

**Pull request merged**